### PR TITLE
#6 feat user generated cards

### LINKED ISSUE #15: - **#6** Adding cue cards
- [x] Add a new feature to the application which provides the user the option to add custom made cue cards. 

Acceptance criteria:
- The user should be able to rate the difficulty of the word between  1 (easy), 2(medium) and 3(hard).
- The user should be able to add a descriptive category to the word (e.g shopping, food or sports).
- The user needs to provide the german word as well as its translation.

### LINKED ISSUE #16: - **#7** Storing data
- [x] Set up local storage to store store changes

### PR DESCRIPTION
**Pull request by JannisHeyer**
- I added a feature allowing the user to add his own cards if the vocabulary doesn't already exists in the array.
- Changes will now be stored in the local storage.
- If the user completes the stack by setting all cards to inactive - a "end card" will now be displayed.
- The vocabulary useState is now located in the _app.jsx and available to all components. 
- Some tweaks based on the last code review.

### CODE REVIEW

File: components/Card.jsx [hunk 1]
```diff
@@ -130,3 +128,22 @@ const StyledButton = styled.button`
...
+const StyledPopup = styled.div`
+  position: relative;
+  text-align: center;
+  height: 70vw;
+  width: 70vw;
+  position: relative;
+  background-color: white;
+  border: var(--card-border);
+  margin-bottom: 10rem;
+  & button {
```
Comment (djfarly) [inline components/Card.jsx]: i think you can get rid of the &
```suggestion
  button {
```

File: pages/createCard.jsx [hunk 2]
```diff
@@ -1,15 +1,122 @@
...
+import { useForm } from "react-hook-form";
+import useFormPersist from "react-hook-form-persist";
+import Popup from "reactjs-popup";
+
+const CreateCard = ({ vocabularies, setVocabularies }) => {
+  const { register, handleSubmit, watch, setValue, reset } = useForm();
+  useFormPersist("form", { watch, setValue });
+  const [newCard, setNewCard] = useState();
+  const [open, setOpen] = useState(false);
+  const closeModal = () => setOpen(false);
```
Comment (djfarly) [inline pages/createCard.jsx:14]: for consistency i'd add a second function for opening. ;) That way the closing function does not feel alone. 
```suggestion
  const closeModal = () => setOpen(false);
  const openModal = () => setOpen(true);
```

**Review by djfarly — APPROVED** [review]
Hej Jannis,

First PR of the course where I'll just say: LGTM. 🥇
(maybe because it's already late 😄) 
 
Comments are just minor improvements. Keep up the great work!

File: pages/createCard.jsx (hunk 2 above)
Comment (JannisHeyer) [reply pages/createCard.jsx:14]: Probably the most wholesome improvement! Closing function will get its buddy immediately 🙂

**Pull request merged**